dali/pipeline/executor/queue_policy.h

```diff
@@ -163,14 +164,20 @@ struct UniformQueuePolicy {
 // Ready buffers from previous stage imply that we can process corresponding buffers from current
 // stage
 struct SeparateQueuePolicy {
-  static bool IsUniformPolicy() {
-    return false;
+  static StageQueues GetQueueSizes(QueueSizes init_sizes) {
+    StageQueues result(0);
+     // For non-uniform case we buffer for CPU x GPU pair.
+    result[OpType::SUPPORT] = init_sizes.cpu_size * init_sizes.gpu_size;
+    result[OpType::CPU] = init_sizes.cpu_size;
+    // Mixed and GPU are bound together due to being outputs
+    result[OpType::MIXED] = init_sizes.gpu_size;
+    result[OpType::GPU] = init_sizes.gpu_size;
+    return result;
   }
 
-  void InitializeQueues(
-      const std::array<int, static_cast<int>(OpType::COUNT)> &stage_queue_depths) {
+  void InitializeQueues(const StageQueues &stage_queue_depths) {
     for (int stage = 0; stage < static_cast<int>(OpType::COUNT); stage++) {
```

Comment:
```
for (auto op_type : {OpType::SUPPORT, OpType::CPU, OpType::MIXED, OpType::GPU } ) {
    for (int i = 0; i < stage_queue_depths[op_type]; i++) {
        // ...
    }
}
```

Reply:
I can omit spelling them out here.

Reply:
I saw that you had that list somewhere else, so I thought that you could reuse that

Reply:
But it was used when I needed to specify the order. Probably better thing to do would be to actually reorder the enum, but I don't know if that wouldn't break something, so if I decide to do that, it would be a separate PR.